### PR TITLE
Add NoSpace subcode to IOError

### PR DESCRIPTION
**Pull request by edouarda**
This will enable users to distinguish "out of disk space" errors from other I/O error and maybe take the appropriate action.

### CODE REVIEW

File: include/rocksdb/status.h [hunk 1]
```diff
@@ -33,14 +33,14 @@ class Status {
   Status& operator=(const Status& s);
   Status(Status&& s)
 #if !(defined _MSC_VER) || ((defined _MSC_VER) && (_MSC_VER >= 1900))
-    noexcept
+      noexcept
```
Comment (siying) [inline include/rocksdb/status.h]: I didn't get this change.

File: include/rocksdb/status.h [hunk 2]
```diff
@@ -58,7 +58,7 @@ class Status {
     kAborted = 10,
     kBusy = 11,
     kExpired = 12,
-    kTryAgain = 13,
+    kTryAgain = 13
```
Comment (siying) [inline include/rocksdb/status.h]: Can you revert this line change? This is unnecessary.

**Comment by siying** [comment]
It mostly looks fine to me. I'll merge if tests pass.

File: include/rocksdb/status.h (hunk 2 above)
Comment (edouarda) [inline include/rocksdb/status.h]: I think it's clang-format doing his work.

File: include/rocksdb/status.h (hunk 1 above)
Comment (edouarda) [inline include/rocksdb/status.h]: clang-format

File: include/rocksdb/status.h (hunk 2 above)
Comment (siying) [inline include/rocksdb/status.h]: It's fine then.

**Comment by edouarda** [comment]
There was a missing include.

**Comment by siying** [comment]
Mac build failed. Can you fix it?

**Comment by edouarda** [comment]
I can fix anything :-)

**Comment by edouarda** [comment]
Not sure the failures are related to this subcode.

**Comment by siying** [comment]
Right it is not related. I'm merging it.

**Pull request merged**